cms/djangoapps/contentstore/views/component.py

```diff
@@ -122,10 +122,33 @@ def container_handler(request, usage_key_string):
 
             component_templates = get_component_templates(course)
             ancestor_xblocks = []
+            parent = get_parent_xblock(xblock)
             action = request.GET.get('action', 'view')
 
             is_unit_page = is_unit(xblock)
-            unit = xblock if is_unit_page else get_parent_xblock(xblock)
+            unit = xblock if is_unit_page else None
+
+            is_first = True
+            block = xblock
```

Comment:
Nit: would be nice to have a comment here, something like "Build the breadcrumbs and find the "unit" ancestor if it's not the immediate parent"

Reply:
Comment added.